components/sup/src/manager/mod.rs

```diff
@@ -145,6 +146,13 @@ impl Manager {
         loop {
             let next_check = SteadyTime::now() + TimeDuration::milliseconds(1000);
             if self.check_for_incoming_signals() {
+                let mut services = self.state
+                    .services
+                    .write()
+                    .expect("Services lock is poisend!");
+                for service in services.drain(..) {
```

Comment:
Yeah, I was really happy to find and use this! Only catch is that the call takes a mutable borrow on self, meaning that if `self.remove(&self)` needed to be `self.remove(&mut self)` we have a problem. I can't remember why I split it off (for sure a compiler suggestion that I wasn't ready to deal with), but I hope we can deal with that when we need to.